### PR TITLE
Fix/storage project copy

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Webserver cannot create a new project from template because storage cancels when copying template data in s3.

### Problem

Storage cancels the handler because the client (i.e. the web-server) times out.  Notice that the cancelation is while creating a client (see [here](https://github.com/ITISFoundation/osparc-simcore/blob/master/services/storage/src/simcore_service_storage/dsm.py#L750)) 

Here we can see relevant part of the logs
- storage: **"POST /v0/simcore-s3/folders". Responding with status 500**
```
ERROR:servicelib.rest_middlewares:Unexpected server error "<class 'concurrent.futures._base.CancelledError'>" from access: 10.3.95.151 "POST /v0/simcore-s3/folders". Responding with status 500
Traceback (most recent call last):
 ...
  File "/home/scu/.venv/lib/python3.6/site-packages/simcore_service_storage/handlers.py", line 355, in create_folders_from_project
    user_id, source_project, destination_project, nodes_map
  File "/home/scu/.venv/lib/python3.6/site-packages/simcore_service_storage/dsm.py", line 757, in deep_copy_project_simcore_s3
    Bucket=self.simcore_bucket_name, Prefix=source_folder
  File "/home/scu/.venv/lib/python3.6/site-packages/aiobotocore/client.py", line 121, in _make_api_call
    operation_model, request_dict, request_context)
...  
  File "/home/scu/.venv/lib/python3.6/site-packages/aiohttp/streams.py", line 604, in read
    await self._waiter
concurrent.futures._base.CancelledError <<<

```
- webserver: **"POST /v0/projects" done in 20.20 secs. Responding with status 500**
```
ERROR: simcore_service_webserver.diagnostics_monitoring:_middleware_handler(110) - Unexpected server error "<class 'aiohttp.web_exceptions.HTTPInternalServerError'>" from access: 10.3.95.28 "POST /v0/projects" done in 20.20 secs. Responding with status 500
Traceback (most recent call last):
 ...
  File "/home/scu/.venv/lib/python3.6/site-packages/simcore_service_webserver/projects/projects_handlers.py", line 122, in create_projects
    await clone_data_coro
  File "/home/scu/.venv/lib/python3.6/site-packages/simcore_service_webserver/storage_api.py", line 44, in copy_data_folders_from_project
    ssl=False,
  File "/home/scu/.venv/lib/python3.6/site-packages/aiohttp/client.py", line 1012, in __aenter__
    self._resp = await self._coro
  File "/home/scu/.venv/lib/python3.6/site-packages/aiohttp/client.py", line 504, in _request
    await resp.start(conn)
  File "/home/scu/.venv/lib/python3.6/site-packages/aiohttp/client_reqrep.py", line 860, in start
    self._continue = None
  File "/home/scu/.venv/lib/python3.6/site-packages/aiohttp/helpers.py", line 596, in __exit__
    raise asyncio.TimeoutError from None
concurrent.futures._base.TimeoutError <<<<<<
```

### Hightlights

- refactors in storage ``create_folders_from_project`` -> ``deep_copy_project_simcore_s3``
   - aiobotocore session is created only once instead of upon every call
- upgrades requirements
- Creates ``tests/performance`` with [Locust](https://locust.io/) . 
   - Currently only enabled for manual testing of this issue
   - Tested with 100 parallel users for 569 requests w/o failures 

## Related issue/s

- address issue detected in the e2e
- Part of PR #2276 but submitted as a separate PR for convenience

## How to test

```command
cd osparc-simcore
make devenv
source .venv/bin/activate
cd services/storage
make install-dev
make tests
```

For manual testing, 
- build and deploy stack
```command
cd osparc-simcore
make build-x
make up-prod
```
- register, login, create a template project and share with everybody 
- set the UUID of this template project in ``test/performance/.env``
```command
cd osparc-simcore/tests/performance
make install-dev
make .env
```
- start locust
```command
make start
```
- go to http://localhost:8089/ and hit start ... analyze ... hit stop
image

and this is the projects of one of the fake users: 
image

### CODE REVIEW

**Review by sanderegg — APPROVED** [review]
pair reviewed. Very nice!

File: services/storage/src/simcore_service_storage/dsm.py [hunk 1]
```diff
@@ -745,94 +754,91 @@ async def deep_copy_project_simcore_s3(
...
         fmds = []
-        async with session.create_client(
-            "s3",
-            endpoint_url=self.s3_client.endpoint_url,
-            aws_access_key_id=self.s3_client.access_key,
-            aws_secret_access_key=self.s3_client.secret_key,
-        ) as client:
+        async with self._create_client_context() as client:
+
+            # step 3: list files first to create fmds
```
Comment (GitHK) [inline services/storage/src/simcore_service_storage/dsm.py:824]:
```suggestion
            # step 3: list files first to create FileMetaData entries
```

File: services/storage/src/simcore_service_storage/dsm.py [hunk 2]
```diff
@@ -940,28 +946,22 @@ async def delete_project_simcore_s3(
...
             await conn.execute(delete_me)
 
-        session = aiobotocore.get_session()
-        async with session.create_client(
-            "s3",
-            endpoint_url=self.s3_client.endpoint_url,
-            aws_access_key_id=self.s3_client.access_key,
-            aws_secret_access_key=self.s3_client.secret_key,
-        ) as client:
+        async with self._create_client_context() as client:
```
Comment (GitHK) [inline services/storage/src/simcore_service_storage/dsm.py:949]: why are new client contests required?

File: tests/performance/Makefile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+.PHONY: install
+install-dev: _check_venv_active  ## installs dependencies
+	# installing requirements
+	pip-sync requirements/dev.txt
+
+
+.PHONY: start
+start: _check_venv_active .env  ## starts locust, a scriptable and scalable performance testing tool
+	# Open http://localhost:8089/
+	locust --locustfile locustfile.py --host http://127.0.0.1:9081  --users 3 --spawn-rate 1
```
Comment (GitHK) [inline tests/performance/Makefile:26]: 👍  nice

**Review by GitHK — APPROVED** [review]
Looks nice. Some questions below.

File: services/storage/src/simcore_service_storage/dsm.py (hunk 2 above)
Comment (pcrespov) [reply services/storage/src/simcore_service_storage/dsm.py:949]: I asked myself the same question but this is an  *aiobotocore client* and that is how they use it https://aiobotocore.readthedocs.io/en/latest/#basic-example. I keep a single instance of the session though...

**Pull request merged**